File: Problemset/0263-Ugly Number/Solution.rb
# @param {Integer} num
# @return {Boolean}
def is_ugly(num)
    if num == 0
        return false
    elsif num == 1
        return true
    elsif num % 2 == 0
        return is_ugly(num / 2)
    elsif num % 3 == 0
        return is_ugly(num / 3)
    elsif num % 5 == 0
        return is_ugly(num / 5)
    else
        return false
    end
end

File: Problemset/0263-Ugly Number/Solution.rs
impl Solution {
    pub fn is_ugly(num: i32) -> bool {
        if num == 0 {
            false
        } else if num == 1 {
            true
        } else if num % 2 == 0 {
            Self::is_ugly(num / 2)
        } else if num % 3 == 0 {
            Self::is_ugly(num / 3)
        } else if num % 5 == 0 {
            Self::is_ugly(num / 5)
        } else {
            false
        }
    }
}
